### PR TITLE
V1.0.0 rc2 patches

### PR DESCRIPTION
**Pull request by runcom**
Close #892 
Close #903 
Fix https://github.com/kubernetes-incubator/cri-o/issues/911
Fix https://github.com/kubernetes-incubator/cri-o/issues/898

Tested this the whole morning, run integration and e2e tests. Integrations are fine, on my local machine I got the usual network failures in e2e but nothing broken afaict so this seems good actually...

@mrunalp @rhatdan @sameo et all, PTAL

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by sameo** [comment]
LGTM

**Comment by mrunalp** [comment]
Did we run e2e locally on this?

**Comment by runcom** [comment]
> Did we run e2e locally on this?

I did, and I got expected failures which worked just fine in CI, if you want to triple check, I think you're the only one with a setup to run them :cry:

**Comment by runcom** [comment]
[retest]

**Comment by rhatdan** [comment]
I kicked off a tests for this manually.

**Comment by rhatdan** [comment]
The "re-test" is not working.

**Comment by runcom** [comment]
Thanks Dan

**Comment by cevich** [comment]
@runcom @rhatdan FWIW this did eventually trigger: https://hurl.corp.redhat.com/0e6a056  I think it took so long b/c the Slave had no executor slots and so the trigger-job itself ended up being queued.

**Comment by mrunalp** [comment]
@runcom Some test failures :/

**Comment by runcom** [comment]
pushed last 2 commits to fix selinux @rhatdan PTAL, I've triggered the CI meanwhile

File: test/testdata/container_config_seccomp.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
 				"user": "system_u",
 				"role": "system_r",
 				"type": "svirt_lxc_net_t",
-				"level": "s0:c4-c5"
+				"level": "s0:c4,c5"
```
Comment (rhatdan) [inline test/testdata/container_config_seccomp.json]: These are actually the same, but this is fine.  SELinux supports ranges c4-c5 as well as comma separated list c4,c5

**Comment by rhatdan** [comment]
[retest]

**Comment by runcom** [comment]
Pushed another commit to fix e2e tests and selinux. Basically, kubernetes can send just `level:s0` but our code is erroring out if any of `level, user, role, type` are empty. This is not the case with docker where you could indeed run with `--security-opt label=level:s0` just fine. Last commit fixes that and add a test also.

**Comment by runcom** [comment]
[retest]

**Comment by runcom** [comment]
Alright, RHEL is still having issues but at least I've fixed fedora and e2e, I'll fix rhel now.

**Comment by runcom** [comment]
Alright, figured it out: selinux policy on rhel doesn't have `container_t` type (https://github.com/projectatomic/container-selinux/blob/RHEL-1.12/container.te) but we set it in our `test/testdata/sandbox_config.json`. Fedora works because we do have it in fedora (https://github.com/projectatomic/container-selinux/blob/master/container.te#L598)

I'm fixing the issue and we should be done

**Comment by runcom** [comment]
Assuming `container_runtime_t` can be used in Fedora, the last commit finally fixes all selinux issues and integration tests in CI should work now (fingers crossed)

**Comment by runcom** [comment]
> Assuming container_runtime_t can be used in Fedora,

just tested it and it works:
```
$ docker run -ti --security-opt label=type:container_runtime_t_fuck busybox sh
# ...
```

**Comment by mrunalp** [comment]
Finally green :)

**Comment by runcom** [comment]
@mrunalp I've squashed the SELinux fixes so tests keep being green in bisect. We can now merge this and cut RC2

**Comment by rhatdan** [comment]
svirt_lxc_net_t would work on both and be correct.  Hopefully in RHEL7.5 we can get container_t into RHEL7.  Needs @lvrabec  to update to selinux-policy to get that done.

**Comment by runcom** [comment]
@rhatdan isn't `container_runtime_t` correct? I see it both works on fedora and RHEL

**Comment by rhatdan** [comment]
container_runtime_t is the label for CRI-O, is it an unconfined domain, so it will work, but container_t/svirt_lxc_net_t is the label to run individual containers with.  It is a confined domain and prevents container breakout.  For this test it probably does not matter.

**Comment by mrunalp** [comment]
Let us switch to svirt_lxc_net_t in the tests to ensure that SELinux is working fine.

**Comment by runcom** [comment]
Ack

**Comment by runcom** [comment]
> Let us switch to svirt_lxc_net_t in the tests to ensure that SELinux is working fine.

pushed change to now use svirt_lxc_net_t (I've also triggered the CI). Let's merge then :)

**Comment by cevich** [comment]
> fingers crossed, I re-kicked the CI from Jenkins directly

Cross your toes as well, then it doesn't hurt so much on contact with Jankenz.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**